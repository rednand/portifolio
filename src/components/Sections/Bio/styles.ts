'use client'

import styled from 'styled-components'

export const Wrapper = styled.main`
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  padding-left: 10vh;
  background: linear-gradient(120deg, #2c2a32 57%, #48d8a4 33%);
  font-family: 'Segoe UI', Roboto, 'Helvetica Neue', sans-serif;
  margin: auto;
  scroll-snap-align: start;
`

export const Title = styled.h1`
  font-size: 4rem;
  line-height: 120%;
  padding: 20px;
  width: 70%;
  color: #fefefe;
  font-weight: 400;
  border-left: 10px solid #48d8a4;
`

export const Description = styled.h3`
  font-size: 2.5rem;
  width: 50%;
  padding: 20px;
  color: #fefefe;
  font-weight: 100;
  border-left: 10px solid #48d8a4;
`

export const Text = styled.p`
  font-size: 1.5rem;
  line-height: 150%;
  width: 50%;
  padding: 20px;
  color: #fefefe;
  font-weight: 100;
  border-left: 10px solid #48d8a4;
`
