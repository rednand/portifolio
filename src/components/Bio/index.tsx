import React from 'react'
import * as Style from './styles'

export default function Bio() {
  return (
    <Style.Wrapper>
      <div style={{ margin: 'auto', display: 'flex', alignItems: 'center' }}>
        <div>
          <Style.Title>Renan de Aguiar Modesto</Style.Title>
          <Style.Description>Desenvolvedor Front-End</Style.Description>
          <Style.Text>
            Desenvolvedor de Sistemas com experiência na área de Desenvolvimento
            Web / Front-end. Trabalho há 3 anos desenvolvendo aplicações web
            para a área de crédito e investimentos utilizando ReactJS, HTML,
            CSS, JavaScript e React Native.
          </Style.Text>
        </div>
      </div>
    </Style.Wrapper>
  )
}
