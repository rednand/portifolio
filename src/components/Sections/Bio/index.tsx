import React from 'react'
import * as Style from './styles'

export default function Bio() {
  return (
    <Style.Wrapper>
      <div style={{ margin: 'auto', display: 'flex', alignItems: 'center' }}>
        <div>
          <Style.Title>Renan de Aguiar Modesto</Style.Title>
          <Style.Description>Desenvolvedor Front-End</Style.Description>
          <Style.Text>
            Desenvolvedor de Sistemas com experiência na área de Desenvolvimento
            Web / Front-end. Trabalho há mais de 3 anos desenvolvendo aplicações
            web para a área de crédito e investimentos utilizando ReactJS, HTML,
            CSS, JavaScript, TypeScript, React Native e NextJS.
          </Style.Text>
        </div>
      </div>
    </Style.Wrapper>
  )
}
