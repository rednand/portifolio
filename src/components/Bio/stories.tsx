import { Meta, StoryObj } from '@storybook/react'
import Bio from '.'

export default {
  title: 'Bio',
  component: Bio,
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: StoryObj = {}
