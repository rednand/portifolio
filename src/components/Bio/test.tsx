import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Bio from '.'

describe('Bio Component', () => {
  it('renders the Bio component correctly', () => {
    const { container } = render(<Bio />)

    expect(
      screen.getByText('lorem ipsium lorem ipsiumlorem')
    ).toBeInTheDocument()

    expect(
      screen.getByText(/Analista de Sistemas com experiência na área/i)
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the background color correctly', () => {
    const { container } = render(<Bio />)

    expect(container.firstChild).toHaveStyle(`
      background: linear-gradient(120deg, #2c2a32 57%, #48d8a4 33%);
    `)
  })
})
