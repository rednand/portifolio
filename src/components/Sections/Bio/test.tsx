import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Bio from '.'

describe('Bio Component', () => {
  it('renders the Bio component correctly', () => {
    const { container } = render(<Bio />)

    expect(
      screen.getByRole('heading', { name: /Renan de Aguiar Modesto/i })
    ).toBeInTheDocument()
    expect(screen.getByText(/Desenvolvedor Front-End/i)).toBeInTheDocument()
    expect(
      screen.getByText(/Desenvolvedor de Sistemas com experiÃªncia/i)
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the background color correctly', () => {
    const { container } = render(<Bio />)

    expect(container.firstChild).toHaveStyle(`
      background: linear-gradient(120deg, #2c2a32 57%, #48d8a4 33%);
    `)
  })
})
