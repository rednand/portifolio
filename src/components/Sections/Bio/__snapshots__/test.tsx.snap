// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio Component renders the Bio component correctly 1`] = `
.c0 {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  padding-left: 10vh;
  background: linear-gradient(120deg, #2c2a32 57%, #48d8a4 33%);
  font-family: 'Segoe UI',Roboto,'Helvetica Neue',sans-serif;
  margin: auto;
  scroll-snap-align: start;
}

.c1 {
  font-size: 4rem;
  line-height: 120%;
  padding: 20px;
  width: 70%;
  color: #fefefe;
  font-weight: 400;
  border-left: 10px solid #48d8a4;
}

.c2 {
  font-size: 2.5rem;
  width: 50%;
  padding: 20px;
  color: #fefefe;
  font-weight: 100;
  border-left: 10px solid #48d8a4;
}

.c3 {
  font-size: 1.5rem;
  line-height: 150%;
  width: 50%;
  padding: 20px;
  color: #fefefe;
  font-weight: 100;
  border-left: 10px solid #48d8a4;
}

<main
  class="c0"
>
  <div
    style="margin: auto; display: flex; align-items: center;"
  >
    <div>
      <h1
        class="c1"
      >
        Renan de Aguiar Modesto
      </h1>
      <h3
        class="c2"
      >
        Desenvolvedor Front-End
      </h3>
      <p
        class="c3"
      >
        Desenvolvedor de Sistemas com experiência na área de Desenvolvimento Web / Front-end. Trabalho há mais de 3 anos desenvolvendo aplicações web para a área de crédito e investimentos utilizando ReactJS, HTML, CSS, JavaScript, TypeScript, React Native e NextJS.
      </p>
    </div>
  </div>
</main>
`;
